core/src/avm2/activation.rs

```diff
@@ -999,9 +1000,27 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
                 _ => self.unknown_op(op),
             };
 
-            if let Err(e) = result {
-                log::error!("AVM2 error: {}", e);
-                return Err(e);
+            if let Err(error) = result {
+                if let Some(body) = method.body() {
+                    for e in body.exceptions.iter() {
+                        if instruction_start >= e.from_offset as usize
+                            && instruction_start < e.to_offset as usize
+                            && e.type_name.0 == 0
+                        // Currently we support only typeless catch clauses
+                        {
+                            // Emulate pushing the exception object
+                            let ws = WString::from_utf8_owned(error.to_string());
+                            let exception = AvmString::new(self.context.gc_context, ws);
+                            self.context.avm2.push(exception);
+
+                            reader.seek_absolute(full_data, e.target_offset as usize);
+                            return Ok(FrameControl::Continue);
```

Comment:
The AVM2 overview says (Section 3.3.6):
`The value and scope stacks in the handling method are cleared before the handler is entered.` .
The value stack might not be entirely true as you stated on discord, but I think that the scope stack should definitely be cleared here.

Reply:
Done, also added AS test for this.